src/chrome/content/code/ApplicableList.js

```diff
@@ -213,9 +213,13 @@ ApplicableList.prototype = {
   },
 
   add_command: function(rule) {
+      // basic validation for data to be added to xul
+      if(!String(rule.id).match(/^[a-zA-Z_0-9]+$/))
```

Comment:
Local style is a space between `if` and `(`, and braces around all blocks.

Also, this would probably be better as:

```
var ruleId = String(rule.id);
if (!ruleId.match(/.../)) {
  return;
}
setAttribute(... ruleId + 'blah')
```

That is, reuse the same variable so it's clearer that it's the one you checked.